lib/eth/EthRpc.js

```diff
@@ -177,56 +179,200 @@ class EthRPC {
+   * @returns {number} The estimated maximum base fee.
+   */
+  async estimateMaxFee({ percentile,  priority }) {
+    const lastBlock = await this.web3.eth.getBlock('latest');    
+    const baseFeePerGas = parseInt(lastBlock.baseFeePerGas) ? parseInt(lastBlock.baseFeePerGas) : 0;
+    const gwei = parseInt(this.web3.utils.toWei('1', 'gwei'));
+
+    if (priority && parseInt(priority) > 0) {
+      // user defined priority fee
+      return 2 * baseFeePerGas + (priority * gwei);
+    }
+    if (percentile && parseInt(percentile) > 0 && parseInt(percentile) <= 100) {
+      // percentile priority fee
+      const maxPriorityFee = await this.estimateMaxPriorityFee({ percentile });
+      return 2 * baseFeePerGas + maxPriorityFee;
+    }
+    // default max fee formmula. ensures inclusion in next block
+    return 2 * baseFeePerGas + (2.5 * gwei); 
+  }
+
+  /**
+   * Estimates the maximum priority fee.
+   * @param {Object} params - The parameters for estimating the maximum priority fee.
+   * @param {number} [params.percentile=25] - The percentile to use for the estimation.
+   * @returns {Promise<number>} The estimated maximum priority fee.
+   */
+  async estimateMaxPriorityFee({ percentile = 25 }) {
+    const rpcEstimate = parseInt(this.web3.utils.toWei('2.5', 'gwei'));
+    const getFees = (_txs) => _txs
+      .filter((tx) => !!tx.maxPriorityFeePerGas)
+      .map((tx) => parseInt(tx.maxPriorityFeePerGas));
+    return this._estimateFee({
+      cache: this.blockMaxPriorityFeeCache,
+      rpcEstimate,
+      percentile,
+      getFees,
+    });
+  }
+
+  /**
+   * Fetches the last n blocks from the blockchain.
+   * @param {Object} params - The parameters for fetching the blocks.
+   * @param {number} [params.n=10] - The number of blocks to fetch.
+   * @returns {Promise<Array>} The fetched blocks.
+   */
+  async _getRecentBlocks({ n = 10 }) {
     const bestBlock = await this.web3.eth.getBlockNumber();
-    const blocks = await Promise.all([...Array(10).keys()]
-      .map((n) => {
-        let targetBlockNumber = bestBlock - n;
-        if (this.blockGasPriceCache.has(targetBlockNumber)) {
-          return this.blockGasPriceCache.get(targetBlockNumber);
+    let cache = this.blockCache;
+    let blocks = await Promise.all(
+      [...Array(n).keys()].map(async (n) => {
+        const targetBlockNumber = bestBlock - n;
+        if (cache && cache.has(targetBlockNumber)) {
+          return cache.get(targetBlockNumber);
         }
-        return this.web3.eth.getBlock(targetBlockNumber, 1);
-      }));
-    let percentile25GasPrices = [];
+        const block = await this.web3.eth.getBlock(targetBlockNumber, 1);
+        if(!block) {
+          throw new Error(`Unable to fetch block ${targetBlockNumber}`);
+        }
+        if (cache) {
+          cache.set(targetBlockNumber, block);
+          if (cache.size > n) {
+            cache.delete(Array.from(cache.keys()).sort()[0]);
```

Comment:
Glad I caught this because it wasn't very obvious but when the sort method isn't given a sort function then its default is to sort each character in its ASCII order. So the problem with that is if I sort [8, 9, 10, 11, 12] the order would be [10, 11, 12, 8 ,9].

The fix is easy enough, Just pass a sorting function like this: .sort((a, b) => a - b)